core/src/main/java/org/mule/lifecycle/RegistryBrokerLifecycleManager.java

```diff
@@ -32,13 +32,12 @@ public RegistryBrokerLifecycleManager(String id, Registry object, MuleContext mu
     @Override
     protected void registerPhases()
     {
-        RegistryLifecycleCallback callback = new RegistryLifecycleCallback();
-        registerPhase(Initialisable.PHASE_NAME, new MuleContextInitialisePhase(),
-            new EmptyLifecycleCallback<AbstractRegistryBroker>());
+        RegistryLifecycleCallback callback = new RegistryLifecycleCallback(this);
+        LifecycleCallback<AbstractRegistryBroker> emptyCallback = new EmptyLifecycleCallback<>();
+        registerPhase(Initialisable.PHASE_NAME, new MuleContextInitialisePhase(), emptyCallback);
         registerPhase(Startable.PHASE_NAME, new MuleContextStartPhase(), callback);
         registerPhase(Stoppable.PHASE_NAME, new MuleContextStopPhase(), callback);
-        registerPhase(Disposable.PHASE_NAME, new MuleContextDisposePhase(),
-            new EmptyLifecycleCallback<AbstractRegistryBroker>());
+        registerPhase(Disposable.PHASE_NAME, new MuleContextDisposePhase(), emptyCallback);
```

Comment:
because registries do their own initialisation and disposal when the muleContext does. This is not something that is very clean on the overall design. However, fixing this was non essential to the end goal so I left it untouched in order to limit the impact of the refactor and decrease risk